test/TestUtils/RuntimeInformationExtensions.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.CoreSetup.Test
+{
+    public static class RuntimeInformationExtensions
+    {
+        public static string GetExeExtensionForCurrentOSPlatform()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return ".exe";
+            }
+            else
+            {
+                return "";
+            }
+        }
+
+        public static string GetSharedLibraryExtensionForCurrentPlatform()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return ".dll";
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return ".dylib";
+            }
+            else
+            {
+                return ".so";
+            }
+        }
+
+        public static string GetSharedLibraryPrefixForCurrentPlatform()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return "";
+            }
+            else
+            {
+                return "lib";
+            }
+        }
+
+        public static string GetDefaultNugetDirectoryForCurrentPlatform()
```

Comment:
fixed. It'd be cool if we had some static analysis tests for dead code.

---
In reply to: [66351367](https://github.com/dotnet/core-setup/pull/97#discussion_r66351367) [](ancestors = 66351367,66347079)